File: easy/Panacea---Truth-Or-Lie/Panacea---Truth-Or-Lie.cs
using System;
using System.IO;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            // do something with line
            ShowVirusStopped(line);
        }
    }
    
    static void ShowVirusStopped(string line){
        int pos = line.IndexOf('|');
        string[] virus = line.Substring(0,pos-1).Trim().Split(' ');
        string[] antvirus = line.Substring(pos+1).Trim().Split(' ');
        int antSum = 0;
        int virSum = 0;
        foreach(string str in virus) virSum += Convert.ToInt32(str, 16);
        foreach(string str in antvirus) antSum += Convert.ToInt32(str, 2);
        Console.WriteLine(antSum>=virSum);
    }
    
    
}

File: easy/Panacea---Truth-Or-Lie/Panacea---Truth-Or-Lie.java
import java.io.*;
public class Main {
    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            ShowVirusStopped(line);
        }
    }
    
    static void ShowVirusStopped(String line){
        int pos = line.indexOf("|");
        String[] virus = line.substring(0,pos-1).trim().split("\\s+");
        String[] antvirus = line.substring(pos+1).trim().split("\\s+");
        int antSum = 0;
        int virSum = 0;
        for(String str : virus) virSum += Integer.parseInt(str, 16);
        for(String str : antvirus) antSum += Integer.parseInt(str, 2);
        System.out.println(antSum>=virSum? "True":"False");
    }
}
